common-lib/src/main/scala/com/gu/mediaservice/lib/elasticsearch/Mappings.scala

```diff
@@ -22,13 +22,13 @@ object Mappings {
       val maximumStringLengthToStore = (maximumBytesOfKeywordInUnderlyingLuceneIndex * .9).toInt
 
       dynamicTemplate("file_metadata_fields_as_keywords").
-        mapping(dynamicKeywordField().index(false).store(true).ignoreAbove(maximumStringLengthToStore)).
+        mapping(dynamicKeywordField().index(true).store(true).ignoreAbove(maximumStringLengthToStore)).
         pathMatch("fileMetadata.*").matchMappingType("string")
     }
 
     def storedJsonObjectTemplate: DynamicTemplateRequest = {
       dynamicTemplate("stored_json_object_template").
-        mapping(dynamicType().index(false).store(true)).
+        mapping(dynamicType().index(true).store(true)).
```

Comment:
It looks like this property can't be updated on an existing field 😞 

```
Mapper for [fileMetadata.icc.Device manufacturer] conflicts with existing mapping:\n[mapper [fileMetadata.icc.Device manufacturer] has different [index] values]
```

`PUT images/_mapping`

```json
{
    "properties": {
        "fileMetadata": {
            "dynamic": "true",
            "properties": {
                "icc": {
                    "type": "object",
                    "properties": {
                        "Device manufacturer": {
                            "type": "keyword",
                            "index": true,
                            "store": true,
                            "ignore_above": 29489
                        }
                    }
                }
            }
        }
    }
}
```

```json

{
  "error": {
    "root_cause": [
      {
        "type": "illegal_argument_exception",
        "reason": "Mapper for [fileMetadata.icc.Device manufacturer] conflicts with existing mapping:\n[mapper [fileMetadata.icc.Device manufacturer] has different [index] values]"
      }
    ],
    "type": "illegal_argument_exception",
    "reason": "Mapper for [fileMetadata.icc.Device manufacturer] conflicts with existing mapping:\n[mapper [fileMetadata.icc.Device manufacturer] has different [index] values]"
  },
  "status": 400
}

```

So I think we need to discuss how to integrate this @sihil?

Reply:
Hello @mbarton.
Are the creation of *file_metadata_fields_as_keywords* and *stored_json_object_template* as in the code exactly the same nature of the test that you did? Because I tried to upload a photo with _index_ flags set to false, then I updated the grid setting the _index_ to true and search for it, it worked without any errors.

Reply:
This mapping only gets used when the index is first created. As such if you later disable it without blowing away the index this will still work as described.

On the flip side, this won't impact us directly as our mappings will still have `index=false` - but the feature won't work as expected.

Reply:
Yep exactly that, I couldn't change the setting after the index had been created. It definitely worked if `index: true` was in the initial mapping, sorry @gribeiro I should have been clearer there

> if you later disable it without blowing away the index this will still work as described

@sihil Did ES let you go from `index: true` to `index: false` on an existing index?

Reply:
No. My point was that if you created the index when `index: true` and then deployed code which has `index: false` over the top without re-creating the index then you would find it was still searchable.

Reply:
I'm open for any suggestions on how we proceed with this issue.
It is possible to use this as we have it right now and you can schedule some re-index of ES later? We could set this as some configuration, we could even let the whole functionality behind this setting -- if the user tries to search for fields that are not indexed, currently he will got some ugly errors on kahuna